### ISSUE TITLE
`--filter` command doesn't work

### ISSUE THREAD

**Issue by xobs** [gimli-rs/ddbug#42]
I just installed 0.4.0 from git with the git command present on the README.

`ddbug` is able to read a file, however when I specify a filter it tells me that `<FILE>` was not provided.

Running `ddbug` on an elf file:
```
bash-3.2$ ddbug elf | head
file elf
        size: 204272
        fn size: 247142
        var size: 70805
        other size: 0
        sections:
                .vector_table
                        address: 0x8000000-0x8000297
                        size: 664
                .text
[2024-11-20T10:36:22Z ERROR ddbug] elf: IO error: Broken pipe (os error 32)
bash-3.2$
```

However, adding `--filter name=foo` does not work:

```
bash-3.2$ ddbug --filter name=foo elf
error: the following required arguments were not provided:
  <FILE>

Usage: ddbug --filter <FILTER>... <FILE>

For more information, try '--help'.
bash-3.2$
```

**Comment by philipc** [comment]
Thanks for the bug report. As a workaround, you can put the file first (`ddbug elf --filter name=foo`).

**Comment by philipc** [comment]
Fixed in ec3cc6d

**Comment by xobs** [comment]
Thanks! That does fix it for me.
